### PR TITLE
Fix version_path in SOURCE links

### LINKED ISSUE #153733: [SOURCE] links for patch releases lead to 2.7.1
### 📚 The doc issue

Here is an example that @stas00 pointed out:

> clicking on [Source] from https://docs.pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm
> takes me to
> https://github.com/pytorch/pytorch/blob/v2.7.1/torch/nn/modules/normalization.py#L94
> which is 404
> it appears that it prematurely links to 2.7.1 which doesn't yet exist
> it should be:
> https://github.com/pytorch/pytorch/blob/v2.7.0/torch/nn/modules/normalization.py#L94

I checked that it is true for all current 2.7 docs. 

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by svekars**
Fixes: #153733

cc @malfet @seemethere @sekyondaMeta @AlannaBurke

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -3484,15 +3486,17 @@ def linkcode_resolve(domain, info):
...
+        if "-rc" in torch_version:
+            # Use the actual git version for RC builds
+            version_path = torch_version
+        else:
+            # For stable releases, use vX.Y.Z format
+            patch_version = version_parts[2].split("+")[0].split("a")[0]
+            version_path = f"v{version_parts[0]}.{version_parts[1]}.{patch_version}"
     else:
         version_path = torch.version.git_version
+    print(f"Version path: {version_path}")
```
Comment (malfet) [inline docs/source/conf.py:3499]: You probably want to remove print?

**Review by malfet — COMMENTED** [review]
And what is the test plan?

File: docs/source/conf.py [hunk 2]
```diff
@@ -3484,15 +3486,17 @@ def linkcode_resolve(domain, info):
...
     if RELEASE:
-        version_parts = torch_version.split(
-            "."
-        )  # For release versions, format as "vX.Y.Z" for correct path in repo
-        patch_version = (
-            version_parts[2].split("+")[0].split("a")[0]
-        )  # assuming a0 always comes after release version in versions.txt
-        version_path = f"v{version_parts[0]}.{version_parts[1]}.{patch_version}"
+        version_parts = torch_version.split(".")
+        if "-rc" in torch_version:
```
Comment (atalman) [inline docs/source/conf.py:3490]: I believe for rc builds ``torch.__version__==2.7.1`` rather then ``2.7.1-rc1``

File: docs/source/conf.py [hunk 3]
```diff
@@ -3484,15 +3486,17 @@ def linkcode_resolve(domain, info):
...
     # Determine the tag based on the torch_version
     if RELEASE:
-        version_parts = torch_version.split(
-            "."
-        )  # For release versions, format as "vX.Y.Z" for correct path in repo
-        patch_version = (
-            version_parts[2].split("+")[0].split("a")[0]
-        )  # assuming a0 always comes after release version in versions.txt
-        version_path = f"v{version_parts[0]}.{version_parts[1]}.{patch_version}"
+        version_parts = torch_version.split(".")
```
Comment (Skylion007) [inline docs/source/conf.py:3489]: Consider using packaging.version
